ReactiveExample/LoginViewModel.cs

```diff
@@ -6,72 +6,78 @@
 
-namespace ReactiveExample {
-	//
-	// This view model can be easily shared across different UI frameworks.
-	// For example, if you have a WPF or XF app with view models written
-	// this way, you can easily port your app to Terminal.Gui by implementing
-	// the views with Terminal.Gui classes and ReactiveUI bindings.
-	//
-	// We mark the view model with the [DataContract] attributes and this
-	// allows you to save the view model class to the disk, and then to read
-	// the view model from the disk, making your app state persistent.
-	// See also: https://www.reactiveui.net../docs/handbook/data-persistence/
-	//
-	[DataContract]
-	public class LoginViewModel : ReactiveObject {
-		readonly ObservableAsPropertyHelper<int> _usernameLength;
-		readonly ObservableAsPropertyHelper<int> _passwordLength;
-		readonly ObservableAsPropertyHelper<bool> _isValid;
-		
-		public LoginViewModel () {
-			var canLogin = this.WhenAnyValue (
-				x => x.Username, 
-				x => x.Password,
-				(username, password) =>
-					!string.IsNullOrEmpty (username) &&
-					!string.IsNullOrEmpty (password));
-			
-			_isValid = canLogin.ToProperty (this, x => x.IsValid);
-			Login = ReactiveCommand.CreateFromTask<EventArgs> (
-				e => Task.Delay (TimeSpan.FromSeconds (1)),
-				canLogin);
-			
-			_usernameLength = this
-				.WhenAnyValue (x => x.Username)
-				.Select (name => name.Length)
-				.ToProperty (this, x => x.UsernameLength);
-			_passwordLength = this
-				.WhenAnyValue (x => x.Password)
-				.Select (password => password.Length)
-				.ToProperty (this, x => x.PasswordLength);
-			
-			Clear = ReactiveCommand.Create<EventArgs> (e => { });
-			Clear.Subscribe (unit => {
-				Username = string.Empty;
-				Password = string.Empty;
-			});
-		}
-		
-		[Reactive, DataMember]
-		public string Username { get; set; } = string.Empty;
-		
-		[Reactive, DataMember]
-		public string Password { get; set; } = string.Empty;
-		
-		[IgnoreDataMember]
-		public int UsernameLength => _usernameLength.Value;
-		
-		[IgnoreDataMember]
-		public int PasswordLength => _passwordLength.Value;
+namespace ReactiveExample;
 
-		[IgnoreDataMember]
-		public ReactiveCommand<EventArgs, Unit> Login { get; }
-		
-		[IgnoreDataMember]
-		public ReactiveCommand<EventArgs, Unit> Clear { get; }
-		
-		[IgnoreDataMember]
-		public bool IsValid => _isValid.Value;
-	}
-}
\ No newline at end of file
+//
+// This view model can be easily shared across different UI frameworks.
+// For example, if you have a WPF or XF app with view models written
+// this way, you can easily port your app to Terminal.Gui by implementing
+// the views with Terminal.Gui classes and ReactiveUI bindings.
+//
+// We mark the view model with the [DataContract] attributes and this
+// allows you to save the view model class to the disk, and then to read
+// the view model from the disk, making your app state persistent.
+// See also: https://www.reactiveui.net../docs/handbook/data-persistence/
+//
+[DataContract]
```

Comment:
Non-review comment I wanted to add about this (but it makes it a review comment now that I've started one):

The JsonSerializer now respects DataContracts!

So, anywhere we currently use certain attributes specific to System.Text.Json can instead be these attributes, which enable broader serialization compatibility and consistency beyond just JSON.